rust/operator-binary/src/airflow_controller.rs

```diff
@@ -1255,13 +1249,26 @@ fn build_executor_template_config_map(
     }
 
     if merged_executor_config.logging.enable_vector_agent {
-        pb.add_container(build_logging_container(
-            resolved_product_image,
-            merged_executor_config
-                .logging
-                .containers
-                .get(&Container::Vector),
-        )?);
+        match airflow
+            .spec
+            .cluster_config
+            .vector_aggregator_config_map_name
+            .to_owned()
+        {
+            Some(vector_aggregator_config_map_name) => {
+                pb.add_container(build_logging_container(
+                    resolved_product_image,
+                    merged_executor_config
+                        .logging
+                        .containers
+                        .get(&Container::Vector),
+                    &vector_aggregator_config_map_name,
```

Comment:
With the above suggestion, this can change to:

```suggestion
                    vector_aggregator_config_map_name,
```